corehq/form_processor/track_related.py

```diff
@@ -0,0 +1,48 @@
+from collections import defaultdict
+
+
+class TrackRelatedChanges(object):
```

Comment:
Not sure if this is overkill or not but I wasn't loving having to do `hasattr(case, 'unsaved_X')` etc. I think we'll need the same thing for attachments as well.